### ISSUE TITLE
Migrate to new terminology in database and generated types

### ISSUE THREAD

**Issue by mattpocock** [mattpocock/sextant#7]
We should work out what best to call things in Sextant. I am keen to hear about prior art, and what feels most natural for you.

## Environments

Environments are things that can fire events and receive events, like actors in an actor system.

![image](https://user-images.githubusercontent.com/28293365/97744317-e6ab9b00-1ade-11eb-8d72-0edd53e03e40.png)

#### Possible Alternatives:

Actors. Resources.

## Events

Events are what link up environments to each other.

![image](https://user-images.githubusercontent.com/28293365/97744254-d4316180-1ade-11eb-87d2-1805831be036.png)

#### Possible alternatives

None. I like 'events'. They match up to 'event-driven-programming', XState 'events'.

## Sequences

A sequence is a chain of events. A single sequence matches a single path through the system, like a success or failure case.

![image](https://user-images.githubusercontent.com/28293365/97744500-40ac6080-1adf-11eb-84fa-fc6a758a32cc.png)

#### Possible alternatives

Test cases. Cases. Chains. Branches. Variants. Paths. Lots here.

## Services

A service is a group of sequences. Together, sequences form an overview of possible paths through a system. You can see what happens when something works, and when it doesn't. This graph of possibilities is the service.

#### Possible alternatives

Systems. Behaviour Graphs...? Workflows. Features. Lots here too.

# EDIT

The new terminology is Features, Scenarios, Actors and Events. We should migrate the database and generated types to use these new terms.

**Comment by ezhikov** [comment]
## Environments

Basically you have either Actors or Lifelines in sequence diagrams. Difference is that Lifeline - internal for system, and Actor is external. But, by the Actor model everything is and actor, so "Actor" fits great. With "Environment" there is lots to process before you guess what does it mean.

Also you can call this "Role", but not sure if it's suitable.

## Events

May be also "Messages", but "Events" is good

## Sequence

Maybe "Scenario" is better, since it can cover your alternatives like "Cases/TestCases" (Test Scenario), and "Path/Branch" (Successful Scenario, Unsuccessful Scenario)

## Services

UML have concept of Interaction. I believe it fits here nicely. You describe different scenarios for single Interaction

## TL;DR;

- Actors
- Events
- Scenarios
- Interactions

In a sentence: "Sextant is a tool to visually model set of _scenarios_ of  _interactions_ between your _actors_ (via sequence of _events_) and generate types for this _interactions_".

**Comment by mattpocock** [comment]
I like actors, events, and I LOVE scenarios. But I'm not convinced on 'interactions'. I agree that it's better than 'Service', but there might be something else.

**Comment by mattpocock** [comment]
I quite like 'Feature', following Gherkin syntax:

https://cucumber.io/docs/gherkin/reference/

**Comment by mattpocock** [comment]
@ezhikov I've been thinking about this more, and I really like:

Service -> Feature
Environment -> Actor
Event -> Event
Sequence -> Scenario

I'm going to run this by my team on Monday and see what they think.

**Comment by mattpocock** [comment]
One not mentioned above is 'Step', which is the ordered steps to conduct a scenario. I like this - it matches Gherkin syntax and feels intuitive.

**Comment by ezhikov** [comment]
I'm feeling neutral about "Feature", and I see absolutely nothing wrong about Step

**Comment by mattpocock** [comment]
@ezhikov OK, I'm set on the above. I'll now change this issue to encompass changing the database and generated types to reflect these new terms.
